wordpress-mobile/WordPress-iOS
# Reader Detail: Animated gif support for featured images

This PR adds: 
1. Animated gif support to the featured image on the reader detail screen
2. Animated gif support to `WPImageViewController` via the new `initWithGifData:` method

![featured_image_gif](https://user-images.githubusercontent.com/154014/39846099-7d8b76a6-53bf-11e8-9993-d3dc0bd887a8.gif)

**Note:** GIF support for the content part of the reader detail screen will come in a subsequent PR.

**Note 2:** This PR also contains a small bugfix in `ReaderPost`'s `contentIncludesFeaturedImage` method — originally it was not finding the featured image in content if the URL contained query string params

Fixes #9262 
Partially addresses #9253 

### Testing

_Try to test this on a variety of devices — iPhones and iPads, old & new_

1. In the reader, find a post that contains an animated gif featured image that is not also in the content 
2. Open the reader post detail screen
3. Verify the animated gif appears (assuming it passes the `LargeGIFPlaybackStrategy` checks...otherwise a static image should appear)
4. Tap on the featured image
5. Verify the animated gif appears in `WPImageViewController` (or a static image if the gif didn't pass the `LargeGIFPlaybackStrategy` check)
6. Try pinch zooming, single tapping (to dismiss), double tapping (to zoom/unzoom), flinging to dismiss, etc

@etoledom would you be so kind to review this?
